File: python/operators.py
"""
PYTHON OPERATORS
This file demonstrates various types of operators in Python:
- Arithmetic operators
- Comparison operators
- Logical operators
- Assignment operators
- Bitwise operators
- Membership operators
- Identity operators
"""

# Arithmetic operators
a = 10
b = 3
print("Arithmetic Operators:")
print(f"{a} + {b} = {a + b}")  # Addition
print(f"{a} - {b} = {a - b}")  # Subtraction
print(f"{a} * {b} = {a * b}")  # Multiplication
print(f"{a} / {b} = {a / b}")  # Division
print(f"{a} // {b} = {a // b}")  # Floor division
print(f"{a} ** {b} = {a ** b}")  # Exponentiation
print(f"{a} % {b} = {a % b}\n")  # Modulus

# Comparison operators
print("Comparison Operators:")
print(f"{a} == {b} is {a == b}")
print(f"{a} != {b} is {a != b}")
print(f"{a} > {b} is {a > b}")
print(f"{a} < {b} is {a < b}")
print(f"{a} >= {b} is {a >= b}")
print(f"{a} <= {b} is {a <= b}\n")

# Logical operators
x = True
y = False
print("Logical Operators:")
print(f"{x} and {y} is {x and y}")
print(f"{x} or {y} is {x or y}")
print(f"not {x} is {not x}\n")

# Assignment operators
print("Assignment Operators:")
c = a
c += b; print(f"c += b → c = {c}")
c -= b; print(f"c -= b → c = {c}")
c *= b; print(f"c *= b → c = {c}")
c /= b; print(f"c /= b → c = {c}")
c //= b; print(f"c //= b → c = {c}")
c **= b; print(f"c **= b → c = {c}")
c %= b; print(f"c %= b → c = {c}\n")

# Bitwise operators
print("Bitwise Operators:")
print(f"{a} & {b} = {a & b}")  # AND
print(f"{a} | {b} = {a | b}")  # OR
print(f"{a} ^ {b} = {a ^ b}")  # XOR
print(f"~{a} = {~a}")  # NOT
print(f"{a} << 1 = {a << 1}")  # Left shift
print(f"{a} >> 1 = {a >> 1}\n")  # Right shift

# Membership operators
list_example = [1, 2, 3, 4, 5]
print("Membership Operators:")
print(f"2 in {list_example} = {2 in list_example}")
print(f"6 not in {list_example} = {6 not in list_example}\n")

# Identity operators
x = 5
y = 5
z = [1, 2, 3]
w = [1, 2, 3]
print("Identity Operators:")
print(f"x is y = {x is y}")  # True (same value, immutable)
print(f"z is w = {z is w}\n")  # False (different objects)

File: c/operators.c
/*
OPERATORS IN C
This file demonstrates various types of operators in C programming:
- Arithmetic operators
- Relational operators
- Logical operators
- Assignment operators
- Bitwise operators
- Ternary operator
*/

#include <stdio.h>

int main() {
    // Arithmetic operators
    int a = 10, b = 5;
    printf("Arithmetic Operators:\n");
    printf("%d + %d = %d\n", a, b, a + b); // Addition
    printf("%d - %d = %d\n", a, b, a - b); // Subtraction
    printf("%d * %d = %d\n", a, b, a * b); // Multiplication
    printf("%d / %d = %d\n", a, b, a / b); // Division
    printf("%d %% %d = %d\n\n", a, b, a % b); // Modulus
    
    // Relational operators
    printf("Relational Operators:\n");
    printf("%d == %d is %d\n", a, b, a == b);
    printf("%d != %d is %d\n", a, b, a != b);
    printf("%d > %d is %d\n", a, b, a > b);
    printf("%d < %d is %d\n", a, b, a < b);
    printf("%d >= %d is %d\n", a, b, a >= b);
    printf("%d <= %d is %d\n\n", a, b, a <= b);
    
    // Logical operators
    int x = 1, y = 0;
    printf("Logical Operators:\n");
    printf("%d && %d is %d\n", x, y, x && y);
    printf("%d || %d is %d\n", x, y, x || y);
    printf("!%d is %d\n\n", x, !x);
    
    // Assignment operators
    printf("Assignment Operators:\n");
    int c = a; // Simple assignment
    c += b; printf("c += b is %d\n", c);
    c -= b; printf("c -= b is %d\n", c);
    c *= b; printf("c *= b is %d\n", c);
    c /= b; printf("c /= b is %d\n", c);
    c %= b; printf("c %%= b is %d\n\n", c);
    
    // Bitwise operators
    printf("Bitwise Operators:\n");
    printf("%d & %d is %d\n", a, b, a & b);  // AND
    printf("%d | %d is %d\n", a, b, a | b);  // OR
    printf("%d ^ %d is %d\n", a, b, a ^ b);  // XOR
    printf("~%d is %d\n", a, ~a);           // NOT
    printf("%d << 1 is %d\n", a, a << 1);   // Left shift
    printf("%d >> 1 is %d\n\n", a, a >> 1); // Right shift
    
    // Ternary operator
    printf("Ternary Operator:\n");
    int max = (a > b) ? a : b;
    printf("Maximum between %d and %d is %d\n", a, b, max);
    
    return 0;
}